components/style_derive/to_animated_zero.rs

```diff
@@ -0,0 +1,79 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use quote;
+use syn;
+use synstructure;
+
+pub fn derive(input: syn::DeriveInput) -> quote::Tokens {
+    let name = &input.ident;
+    let (impl_generics, ty_generics, where_clause) = input.generics.split_for_impl();
+    let mut where_clause = where_clause.clone();
+    for param in &input.generics.ty_params {
+        where_clause.predicates.push(
+            where_predicate(syn::Ty::Path(None, param.ident.clone().into())),
+        );
+    }
+
+    let to_body = match_body(&input);
+
+    quote! {
+        impl #impl_generics ::values::animated::ToAnimatedZero for #name #ty_generics #where_clause {
+            #[allow(unused_variables)]
+            #[inline]
+            fn to_animated_zero(&self) -> Result<Self, ()> {
+                match *self {
+                    #to_body
+                }
+            }
+        }
+    }
+}
+
+fn match_body(input: &syn::DeriveInput) -> quote::Tokens {
```

Comment:
It'd be nice to have a few more comments of what and how this generates the right code, since I haven't used synstructure before I can only guess, though this looks reasonable.